dashboard/src/client/client.ts

```diff
@@ -188,7 +186,24 @@ export class MatterClient {
     if (event.event === "node_removed") {
       delete this.nodes[event.data];
       this.nodes = { ...this.nodes };
-      console.log("node removed!")
+      this.fireEvent("nodes_changed");
+      return;
+    }
+
+    if (event.event === "node_updated") {
+      const node = new MatterNode(event.data);
+      this.nodes = { ...this.nodes, [node.node_id]: node };
+      this.fireEvent("nodes_changed");
+      return;
+    }
+
+    if (event.event === "attribute_updated") {
+      const nodeId = (event.data as Array<any>)[0] as number;
+      const attributeKey = (event.data as Array<any>)[1] as string;
+      const attributeValue = (event.data as Array<any>)[2] as any;
+      const node = new MatterNode(this.nodes[nodeId]);
```

Comment:
Isn't `this.nodes[nodeId]` already an instance of `MatterNode`? As you do save it back to `this.nodes` below?

Reply:
It was the only way I could get reactivity to respond but tell me if you have better ideas

Reply:
`nodes` is the reactive part right? You already make a copy of that